modules/local/extract_sequences.nf

```diff
@@ -0,0 +1,30 @@
+process EXTRACT_SEQUENCES {
+    label 'process_nompi'
+
+    conda (params.enable_conda ? "conda-forge::seqtk=1.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqtk:1.3--h5bf99c6_3' :
+        'quay.io/biocontainers/seqtk:1.3--h7132678_4' }"
+
+    input:
+    tuple val(meta), path(fasta) 
+    path(fai)
+
+    output:
+    tuple val(meta), path("*subsequences.fasta"), emit: subseq
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = meta.prefix ?: ''
+    """
+    cat $fai | cut -f1 >  subseqs.lst
```

Comment:
For this step use this module: https://github.com/nf-core/modules/blob/master/modules/nf-core/cat/cat/main.nf
We want to maintain as little code as possible and using `nf-core/modules` helps us do that.